### ISSUE TITLE
B::CC can't seem to find B::FAKEOP::_save_common_middle()

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#8625]
Migrated from [rt.perl.org#40435](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=40435) (status was 'resolved')

Searchable as RT40435$

**Comment by p5pRT** [comment]
### From fromperlbug.16.hfri@spamgourmet.com

#### Created by fromperlbug.16.hfri@spamgourmet.com



When trying to compile a simple "Hello World"\-type script with B&#8203;::CC
invoked as perl \-MO=CC\, I keep getting the error message

Can't locate object method "\_save\_common\_middle" via package "B&#8203;::FAKEOP"
at /usr/lib/perl5/5\.8\.7/i586\-linux\-thread\-multi/B/C\.pm line 389\.
CHECK failed\-\-call queue aborted\.

In contrast\, B&#8203;::C invoked as perl \-MO=C works and produces a \.c output
file\.

This happens on all of&#8203;:
\- The perl\-5\.8\.7\-5 that came with SuSE Linux 10\.0 OSS
&nbsp;  \(on VMware Server 1\.0\.1\)
\- Active Perl 5\.8\.7 build 815 on Windows
&nbsp;  \(on Windows XP Service Pack 2\)
\- Active Perl 5\.8\.8 build 819 on Windows
&nbsp;  \(on Windows XP Service Pack 2\)

<details><summary>Perl Info</summary>

```

Flags:
    category=library
    severity=low

This perlbug was built using Perl v5.8.7 - Fri Sep  9 14:57:31 UTC 2005
It is being executed now by  Perl v5.8.7 - Fri Sep  9 14:52:52 UTC 2005.

Site configuration information for perl v5.8.7:

Configured by abuild at Fri Sep  9 14:52:52 UTC 2005.

Summary of my perl5 (revision 5 version 8 subversion 7) configuration:
  Platform:
    osname=linux, osvers=2.6.12, archname=i586-linux-thread-multi
    uname='linux smetana 2.6.12 #1 smp wed mar 23 21:52:37 utc 2005 i686
athlon i386 gnulinux '
    config_args='-ds -e -Dprefix=/usr -Dvendorprefix=/usr
-Dinstallusrbinperl -Dusethreads -Di_db -Di_dbm -Di_ndbm -Di_gdbm
-Duseshrplib=true -Doptimize=-O2 -march=i586 -mtune=i686
-fmessage-length=0 -Wall -D_FORTIFY_SOURCE=2 -g -Wall -pipe'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=define use5005threads=undef useithreads=define
usemultiplicity=define
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -DTHREADS_HAVE_PIDS
-DDEBUGGING -fno-strict-aliasing -pipe -D_LARGEFILE_SOURCE
-D_FILE_OFFSET_BITS=64',
    optimize='-O2 -march=i586 -mtune=i686 -fmessage-length=0 -Wall
-D_FORTIFY_SOURCE=2 -g -Wall -pipe',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -DTHREADS_HAVE_PIDS -DDEBUGGING
-fno-strict-aliasing -pipe'
    ccversion='', gccversion='4.0.2 20050901 (prerelease) (SUSE Linux)',
gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t',
lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =''
    libpth=/lib /usr/lib /usr/local/lib
    libs=-lnsl -ldl -lm -lcrypt -lutil -lpthread -lc
    perllibs=-lnsl -ldl -lm -lcrypt -lutil -lpthread -lc
    libc=/lib/libc-2.3.5.so, so=so, useshrplib=true, libperl=libperl.so
    gnulibc_version='2.3.5'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E
-Wl,-rpath,/usr/lib/perl5/5.8.7/i586-linux-thread-multi/CORE'
    cccdlflags='-fPIC', lddlflags='-shared'

Locally applied patches:
    


@INC for perl v5.8.7:
    /usr/lib/perl5/5.8.7/i586-linux-thread-multi
    /usr/lib/perl5/5.8.7
    /usr/lib/perl5/site_perl/5.8.7/i586-linux-thread-multi
    /usr/lib/perl5/site_perl/5.8.7
    /usr/lib/perl5/site_perl
    /usr/lib/perl5/vendor_perl/5.8.7/i586-linux-thread-multi
    /usr/lib/perl5/vendor_perl/5.8.7
    /usr/lib/perl5/vendor_perl
    .


Environment for perl v5.8.7:
    HOME=/home/oracle
    LANG=en_US.UTF-8
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
 
PATH=/home/oracle/bin:/usr/local/bin:/usr/bin:/usr/X11R6/bin:/bin:/usr/g
ames:/opt/gnome/bin:/opt/kde3/bin:/usr/lib/mit/bin:/usr/lib/mit/sbin
    PERL_BADLANG (unset)
    SHELL=/bin/bash


```
</details>

**Comment by p5pRT** [comment]
### From @rgs

The C and CC compiler backends are considered obsolescent and will be
removed in perl 5\.10\.0\, so it's unlikely this bug will get fixed\, given
the lack of volunteers to work on B&#8203;::CC and the availaibility of other
\(better\) solutions to ship self\-contained perl programs \(like PAR\)\.

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
@rgs - Status changed from 'open' to 'stalled'

**Comment by p5pRT** [comment]
### From fromperlbug.16.hfri@spamgourmet.com

Hi Rafael \-\-
 
Thank you very much for your reply\.
 
Please be advised that I did not even actually need B&#8203;::CC as such but stumbled on this apparent problem while looking for something else\, so I thought I'd let you know just in case B&#8203;::CC was supposed to work but someone missed that little malfunction\.
 
Please feel free to disregard my bug report \#40435 if the B&#8203;::CC module is considered obsolescent anyway\.
 
\(By the way\, the thing I was looking for turned out to be B&#8203;::Lint with the undefined\-subs option\, so I think I won't be needing B&#8203;::CC\.\)
 
Sincerely\,
Holger Friedrich\.
 
\-\-
Holger Friedrich
Düsseldorf\, Germany
mailto&#8203;:fromperlbug\.16\.hfri@&#8203;spamgourmet\.com

\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_

Von&#8203;: Rafael Garcia\-Suarez via RT \- perlbug\-followup@&#8203;perl\.org \[mailto&#8203;:\+fromperlbug\+hfri\+f9d1f7007d\.perlbug\-followup\#perl\.org@&#8203;spamgourmet\.com\]
Gesendet&#8203;: Mo 30\.10\.2006 15&#8203;:11
An&#8203;: fromperlbug\.16\.hfri@&#8203;spamgourmet\.com
Betreff&#8203;: \[SPAM\] \- \[perl \#40435\] B&#8203;::CC can't seem to find B&#8203;::FAKEOP&#8203;::\_save\_common\_middle\(\) \(fromperlbug&#8203;: message 2 of 16\) \- Bayesian Filter detected spam



The C and CC compiler backends are considered obsolescent and will be
removed in perl 5\.10\.0\, so it's unlikely this bug will get fixed\, given
the lack of volunteers to work on B&#8203;::CC and the availaibility of other
\(better\) solutions to ship self\-contained perl programs \(like PAR\)\.

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'stalled' to 'open'

**Comment by p5pRT** [comment]
### From @rurban

This fix is to add a @&#8203;B&#8203;::FAKEOP&#8203;::ISA =\('B&#8203;::OP'\);
This ticket should be moved to rt\.cpan\.org\, and I'm planning to take
over B&#8203;::C as module\.

The patch for this problem goes roughly like this&#8203;:
B/C\.pm&#8203;:
@&#8203;@&#8203; \-379\,6 \+420\,10 @&#8203;@&#8203;
&nbsp;     savesym\($op\, "&op\_list\[$ix\]"\);
 \}

\+package B&#8203;::FAKEOP;
\+
\+our @&#8203;ISA = qw\(B&#8203;::OP\);
\+
 sub B&#8203;::FAKEOP&#8203;::new \{
&nbsp;     my \($class\, %objdata\) = @&#8203;\_;
&nbsp;     bless \\%objdata\, $class;
@&#8203;@&#8203; \-402\,6 \+447\,8 @&#8203;@&#8203;
 sub B&#8203;::FAKEOP&#8203;::flags \{ $\_\[0\]\->\{flags\} || 0 \}
 sub B&#8203;::FAKEOP&#8203;::private \{ $\_\[0\]\->\{private\} || 0 \}

\+package B&#8203;::C;
\+


\-\- 
Reini Urban

**Comment by p5pRT** [comment]
p5p@spam.wizbit.be - Status changed from 'open' to 'resolved'
